Makefile

```diff
@@ -2,19 +2,19 @@
 all: buildroot/Makefile buildroot/.config
 	make -C buildroot
 
-clean:
-	BR2_EXTERNAL=.. make -C buildroot clean
-
 buildroot/Makefile:
 	git submodule update --init buildroot
 
 buildroot/.config:
 	BR2_EXTERNAL=.. make -C buildroot fischertechnik_TXT_defconfig
 
+zipdir = build
+
 release:
-	if [ ! -f $(imagedir)/rootfs.img ]; then make; fi
 	$(eval version := $(shell cat buildroot/output/target/etc/fw-ver.txt))
-	$(eval zipfile := ftcommunity-txt-$(version).zip)
+	$(eval zipfile := $(zipdir)/ftcommunity-txt-$(version).zip)
 	$(eval imagedir := buildroot/output/images)
+	if [ ! -f $(imagedir)/rootfs.img ]; then make; fi
+	mkdir -p $(zipdir)
```

Comment:
Slightly better (now it's picking up the image, at least), but still fragile (e.g. what happens if rootfs.img is up to date, but uImage isn't?).

And recursively calling "make" from a shell script line in a makefile still is quirky as hell.

Actually, I don't really see the point of having a conditional build here at all instead of always forcing a rebuild. At best, you save some seconds in case you already did a "make" before the "make release", but that's not worth the added fragility and complexity in my opinion.

My recommendation: simply use 
``` 
imagedir := buildroot/output/images
release: all
	$(eval version := $(shell cat buildroot/output/target/etc/fw-ver.txt))
	$(eval zipfile := ftcommunity-txt-$(version).zip)
	mkdir -p build
	rm -f build/$(zipfile)
	zip -j build/$(zipfile) $(imagedir)/am335x-kno_txt.dtb $(imagedir)/rootfs.img $(imagedir)/uImage
```
instead